packages/compiler-cli/test/compliance/test_runner.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+export type CompilationModeRunner = (compilationModel: 'application'|'library') => void;
+
+export interface RunAllCompilationModesFn {
+  (callback: CompilationModeRunner): void;
+  application(callback: CompilationModeRunner): void;
+  library(callback: CompilationModeRunner): void;
+}
+
+function runAllCompilationModesFn(callback: CompilationModeRunner) {
+  runCompilationModel('application', callback, false);
+  runCompilationModel('library', callback, false);
+}
+
+function runCompilationModel(
+    compilationModel: 'application'|'library', callback: CompilationModeRunner, error: boolean) {
+  describe(`<<CompilationModel: ${compilationModel}>>`, () => {
+    callback(compilationModel);
+    if (error) {
+      afterAll(() => {
+        throw new Error(`runCompilationModel limited to ${compilationModel}, cannot pass`);
+      });
+    }
+  });
+}
+
+export const runAllCompilationModels = runAllCompilationModesFn as RunAllCompilationModesFn;
```

Comment:
How about

```suggestion
export const runForEachCompilationMode = runForEachCompilationModeFn as RunAllCompilationModesFn;
```

(similar to the FileSystem `runInEachFileSystem()`):

For example:
```ts
runForEachCompilationMode(compilationModel => {
  ...
});
```

Reply:
Why not `runInEachCompilationMode()`? :grin:

Reply:
I did go with `runInEachCompilationMode`, just for consistency with the filesystem equivalent.